orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py

```diff
@@ -0,0 +1,1475 @@
+                {
+                    'world_rank' : world_rank,
+                    'world_size' : world_size,
+                    'allreduce_post_accumulation' : True,
+                    'deepspeed_zero_optimization':
+                    {
+                        'stage': 1
+                    }
+                },
+                'debug' : {'deterministic_compute': True}
+            }
+    
+    # extract state dictionaries to compare
+    state_dict_post_checkpoint, model = load_evaluate_extract_state_dict(device, orttrainer.ORTTrainerOptions(opts), checkpoint_dir)
+    state_dict_post_checkpoint = split_state_dict(state_dict_post_checkpoint)
+
+    state = pickle.load(open(checkpoint_dir+'state_dict_'+str(world_rank)+'.pkl', 'rb'))
+    state_dict_pre_checkpoint = split_state_dict(state['state_dict_'+str(world_rank)])
+
+    # compare all states
+    # model states
+    for key, value in state_dict_pre_checkpoint['fp16_param'].items():
+        assert_allclose(value, state_dict_post_checkpoint['fp16_param'][key])
+
+    # compare optimizer states
+    for key, value in state_dict_pre_checkpoint['optimizer'].items():
+        assert_allclose(value, state_dict_post_checkpoint['optimizer'][key])
+
+@distributed_setup
+def test_load_from_distributed_zero_full_precision_into_distributed_zero_mixed_precision(world_rank, world_size, device, checkpoint_dir = 'checkpoint_dir/distributed_zero/full_precision/'):
+    opts = {
+                'device' : {'id' : device},
+                'mixed_precision':
+                {
+                    'enabled': True,
+                    'loss_scaler': amp.DynamicLossScaler()
+                },
+                'distributed' :
+                {
+                    'world_rank' : world_rank,
+                    'world_size' : world_size,
+                    'allreduce_post_accumulation' : True,
+                    'deepspeed_zero_optimization':
+                    {
+                        'stage': 1
+                    }
+                },
+                'debug' : {'deterministic_compute': True}
+            }
+    
+    # extract state dictionaries to compare
+    state_dict_post_checkpoint, model = load_evaluate_extract_state_dict(device, orttrainer.ORTTrainerOptions(opts), checkpoint_dir)
+    state_dict_post_checkpoint = split_state_dict(state_dict_post_checkpoint)
+
+    state = pickle.load(open(checkpoint_dir+'state_dict_'+str(world_rank)+'.pkl', 'rb'))
+    state_dict_pre_checkpoint = split_state_dict(state['state_dict_'+str(world_rank)])
+
+    # compare all states
+    # model states
+    for key, value in state_dict_post_checkpoint['fp16_param'].items():
+        assert_allclose(value, state_dict_pre_checkpoint['fp32_param'][key[:-5]], atol=1e-3)
+
+    # compare optimizer states
+    for key, value in state_dict_pre_checkpoint['optimizer'].items():
+        assert_allclose(value, state_dict_post_checkpoint['optimizer'][key])
+
+if __name__ == '__main__':
```

Comment:
Do we need more than one entry point for testing?

Reply:
I don't think we would have more than 1 entry point. I can remove the ```__main__``` check.

Reply:
Removed ```__main__``` check for all files other than the main entry point (```orttraining_test_checkpoint.py```). I was contemplating whether or not to remove from there as well, and decided to let that one be there.

Reply:
Updated ```orttraining_test_checkpoint.py``` file as well.